include/espi.h

```diff
@@ -0,0 +1,769 @@
+ */
+struct espi_callback {
+	/** This is meant to be used in the driver only */
+	sys_snode_t node;
+
+	/** Actual callback function being called when relevant. */
+	espi_callback_handler_t handler;
+
+	/** An event which user is interested in, if 0 the callback
+	 * will never be called. Such evt_mask can be modified whenever
+	 * necessary by the owner, and thus will affect the handler being
+	 * called or not.
+	 */
+	enum espi_bus_event evt_type;
+};
+/** @endcond */
+
+/**
+ * @cond INTERNAL_HIDDEN
+ *
+ * eSPI driver API definition and system call entry points
+ *
+ * (Internal use only.)
+ */
+typedef int (*espi_api_config)(struct device *dev, struct espi_cfg *cfg);
+typedef bool (*espi_api_get_channel_status)(struct device *dev,
+						enum espi_channel ch);
+/* Logical Channel 0 APIs */
+typedef int (*espi_api_read_request)(struct device *dev,
+					struct espi_request_packet);
+typedef int (*espi_api_write_request)(struct device *dev,
+					struct espi_request_packet);
+/* Logical Channel 1 APIs */
+typedef int (*espi_api_send_vwire)(struct device *dev,
+					enum espi_vwire_signal vw,
+					u8_t level);
+typedef int (*espi_api_receive_vwire)(struct device *dev,
+					enum espi_vwire_signal vw,
+					u8_t *level);
+/* Logical Channel 2 APIs */
+typedef int (*espi_api_send_oob)(struct device *dev,
+					struct espi_oob_packet);
+typedef int (*espi_api_receive_oob)(struct device *dev,
+					struct espi_oob_packet);
+/* Logical Channel 3 APIs */
+typedef int (*espi_api_flash_read)(struct device *dev,
+					struct espi_flash_packet);
+typedef int (*espi_api_flash_write)(struct device *dev,
+					struct espi_flash_packet);
+typedef int (*espi_api_flash_erase)(struct device *dev,
+					struct espi_flash_packet);
+/* Callbacks and traffic intercept */
+typedef int (*espi_api_manage_callback)(struct device *dev,
+					struct espi_callback *callback,
+					bool set);
+
+struct espi_driver_api {
+	espi_api_config config;
+	espi_api_get_channel_status  get_channel_status;
+	espi_api_read_request read_request;
+	espi_api_write_request write_request;
+	espi_api_send_vwire send_vwire;
+	espi_api_receive_vwire receive_vwire;
+	espi_api_send_oob send_oob;
+	espi_api_receive_oob receive_oob;
+	espi_api_flash_read flash_read;
+	espi_api_flash_write flash_write;
+	espi_api_flash_erase flash_erase;
+	espi_api_manage_callback manage_callback;
+};
+/**
+ * @endcond
+ */
+
+
+/**
+ * @brief Configure operation of a eSPI controller.
+ *
+ * This routine provides a generic interface to override eSPI controller
+ * capabilities.
+ *
+ * If this eSPI controller is acting as slave, the values set here
+ * will be discovered as part through the GET_CONFIGURATION command
+ * issued by the eSPI master during initialization
+ *
+ * If this eSPI controller is acting as master, the values set here
+ * will be used by eSPI master to determine minimum common capabilities with
+ * eSPI slave then send via SET_CONFIGURATION command
+ *
```

Comment:
Please indicate in the drawing the flows for both scenarios.  Maybe add a row indicating the Roles of the eSPI controller.

Reply:
Flows already were already indicated, added clarifications and indication when eSPI driver programs the corresponding eSPI controller